File: python/FizzBuzz.py
def divides(n, by):
    return n % by == 0


def fizz_or_buzz(n):
    if divides(n, 15):
        return "fizzbuzz"
    elif divides(n, 5):
        return "buzz"
    elif divides(n, 3):
        return "fizz"
    else:
        return n


def fizzbuzz(n):
    [print(fizz_or_buzz(i)) for i in range(n)]


fizzbuzz(100)

File: ruby/fizzbuzz.rb
def fizzbuzz(number)

  def divides(n, by)
    n % by == 0
  end

  def fizz_or_buzz(n)
    if divides(n, 15)
      'fizzbuzz'
    elsif divides(n, 5)
      'buzz'
    elsif divides(n, 3)
      'fizz'
    else
      n
    end
  end

  (1..number).each { |n|
    puts fizz_or_buzz(n)
  }
end

fizzbuzz 100